packages/x-data-grid/src/utils/cellBorderUtils.ts

```diff
@@ -0,0 +1,31 @@
+import { GridPinnedColumnPosition } from '../hooks/features/columns/gridColumnsInterfaces';
+
+export const shouldCellShowRightBorder = ({
+  pinnedPosition,
+  indexInSection,
+  sectionLength,
+  showCellVerticalBorderRootProp,
+}: {
```

Comment:
I have some hesitations about the object param here. These two functions are called each time a cell renders, if we could avoid creating those 2 objects it would be nice.

Reply:
Also applies to a lesser extent to `getPinnedCellOffset`, that one is called on rows instead of cells.

Reply:
Done